### PR TITLE
Pattern ordering

### PR DESCRIPTION
**Pull request by calebeby**
Closes https://github.com/cloudfour/drizzle/issues/66

This addresses the ordering issue that @tylersticka had, by adding the ability to send patterns to the end of a collection. Negative values go to the beginning of the collection, items without values come next, and items with positive values go to the end.

I updated the pattern ordering fixture with a negative `order` property.

Also, this is a breaking change. This means we will need a new major release (but we are currently `0.0.10` 🤔. Does this mean we should release `1.0.0`?)

Also I haven't addressed this:

> Because we now have helpers that can sort by any property, we might want to focus on creating a common sorting function that possesses the logic to infer default values (e.g. 0) and that isn't limited to just properties named "order".

### CODE REVIEW

File: test/fixtures/patterns/components/button/yet-another.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 name: Something Yet Again
+order: -1
 ---
 
-Hmmm, what goes here.
+Hmmm, what goes here.
```
Comment (tylersticka) [inline test/fixtures/patterns/components/button/yet-another.html]: I still don't fully understand why GitHub yells at us when there isn't a newline at the end of the file, but this red arrow makes me sad...

screen shot 2018-07-02 at 2 07 09 pm

**Comment by tylersticka** [comment]
> Also, this is a breaking change. This means we will need a new major release (but we are currently 0.0.10 🤔. Does this mean we should release 1.0.0?)

Maybe! What do you think, @gerardo-rodriguez?

File: test/fixtures/patterns/components/button/yet-another.html (hunk 1 above)
Comment (calebeby) [reply test/fixtures/patterns/components/button/yet-another.html]: ✔️

**Comment by gerardo-rodriguez** [comment]
> Does this mean we should release 1.0.0?

> Maybe! What do you think, @gerardo-rodriguez?

@tylersticka @calebeby Yep, if it's a breaking change it should be `1.0.0`. 😉

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = false
+insert_final_newline = true
```
Comment (gerardo-rodriguez) [inline .editorconfig]: Can we add a newline in this file? :)

Comment (calebeby) [reply .editorconfig]: Oh haha ✔️

**Pull request merged**